### ISSUE TITLE
Web - Chat - Tiny dots displayed in link name for message/link [([some])](https://google.com)

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19646]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue found when executing PR** https://github.com/Expensify/App/pull/19216

## Action Performed:
1. Login to NewDot
2. open any chat
3. Send a message [([some])](https://google.com/)

## Expected Result:
Link name should be properly displayed

## Actual Result:
Link name has tiny dots y each side

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.18.1

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://platform.applause.com/services/links/v1/external/e3fdf4ff0f7cba80fbc71cca478f3bd9258dc24d7581d2e73445d43a8de640fe

https://platform.applause.com/services/links/v1/external/71f0c644a8455e916d3e34302c5cbee88f362afa2462b29c500071edbee95ba4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by hungvu193** [comment]
that's dot is actually the text decoration line, because by default browsers will “skip” areas where the “ink” of a character crosses an underline. We can fix this issue by adding the `text-decoration-skip-ink: none;` to our `a` tag but this property is not really supportive all the browsers so I'm not sure if we should fix this issue.
<img width="1265" alt="Screenshot 2023-05-26 at 13 56 44" src="https://github.com/Expensify/App/assets/16502320/4a25e772-c0dd-45bb-8d1b-11390899bac8">

[Browser compatibility](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration-skip#browser_compatibility)

**Comment by spcheema** [comment]
Dots appear because of text decoration line `text underline` is cut by special characters due to their height.  It won't be fixed with  `text-decoration-skip-ink: none;` since characters like `j` will appears underneath the underline giving the same dot impressions Screenshot attached. Apologies @hungvu193 

But It can fixed with `text-underline-position: under` css property.  This way browser's rendering engine will take care of it. It's supported by all latest browser butt I'm not sure about the minimum support version SLA.


Support browser version:

* https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration-line
* https://caniuse.com/?search=text-underline-position

![text-decoration-skip-ink - Screenshot 2023-05-26 at 5 06 41 pm](https://github.com/Expensify/App/assets/19812199/4157b01f-d042-4acb-bf6e-b63a08585f71)

![text-underline-position: under Screenshot 2023-05-26 at 5 09 49 pm](https://github.com/Expensify/App/assets/19812199/e9a2ecd8-2a59-4a03-8728-93a801d41d0c)

![Screenshot 2023-05-26 at 5 12 27 pm](https://github.com/Expensify/App/assets/19812199/7d2523de-3612-44e0-9f6a-b7d6c66e7a4e)

**Comment by michaelhaxhiu** [comment]
Weird, let's fix it. Should be low effort and already got some proposals above.

**Comment by towkir** [comment]
@michaelhaxhiu is this available for external contributors?
If so, do I have to submit a new proposal or can work / submit PR based on the existing ones?
Thanks

**Comment by narefyev91** [comment]
@michaelhaxhiu @PauloGasparSv proposal of @spcheema looks good to me

**Comment by michaelhaxhiu** [comment]
@towkir check out this page for guidance on how it works to propose a solution for a job. Contributor only work on a PR after being selected for their proposal. 

https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md

**Comment by spcheema** [comment]
Hi @michaelhaxhiu I have applied for the job on upwork yesterday. I will have  PR ready in soon (ideally 1-2 days)

**Comment by spcheema** [comment]
Contributor details
Your Expensify account email: mannecheema@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01f7b921c5dd619819

**Comment by spcheema** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/19646#issuecomment-1563916118)

**Comment by spcheema** [comment]
@narefyev91  I found that this issue exists in all platform. Unfortunately iOS does not support these properties properties yet. Even iOS mail not render it properly. See attached screenshot.
![IMG_2D24C5CD4FC8-1](https://github.com/Expensify/App/assets/19812199/36f17524-591f-43bc-b7d3-8f467d5d6741)

**Comment by spcheema** [comment]
Linking the iOS device related comment here https://github.com/Expensify/App/pull/20147#issuecomment-1576641949

**Comment by Pujan92** [comment]
Seems a regression issue from this PR. [Slack link](https://expensify.slack.com/archives/C049HHMV9SM/p1686163862186549) 

![Simulator Screen Shot - iPhone 14 - 2023-06-08 at 00 16 34](https://github.com/Expensify/App/assets/14358475/42678889-6a73-4595-a197-6fa091e9e2da)

**Comment by spcheema** [comment]
@Pujan92 it's warning (not error) from react native that `text` property isn't supported since anchor is rendering as text element on iOS platform. That's why this wasn't fixed is IOS devices. As explained here https://github.com/Expensify/App/pull/20147#issuecomment-1577823399

**Comment by spcheema** [comment]
@spcheema @narefyev91 @PauloGasparS otherwise we can exclude styles for iOS explicitly.

**Comment by Pujan92** [comment]
@spcheema I think this needs to be resolved as we usually avoid warnings too. For the cursor prop warnings solution, we do have separate style files.

**Comment by PauloGasparSv** [comment]
> @spcheema I think this needs to be resolved as we usually avoid warnings too. For the cursor prop warnings solution, we do have separate style files.

I agree, thks for the help here @Pujan92! I don't think we caught that warning in reviews : )

> we can exclude styles for iOS explicitly but when these properties are supported in future we need to raise a new PR.

@spcheema I think that's the way to go! Could you please create a follow up P.R. that fixes that warning on iOS?

**Comment by spcheema** [comment]
@PauloGasparSv PR is on the way. Testing android right now. We may have to exclude `Android` along with `iOS`

**Comment by narefyev91** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/20462
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment - https://github.com/Expensify/App/issues/19646#issuecomment-1581548728
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: https://expensify.slack.com/archives/C049HHMV9SM/p1686163862186549
- Determine if we should create a regression test for this bug - no needs - it was mostly clean up for warnings

**Comment by PauloGasparSv** [comment]
> @PauloGasparSv I am still new to the process. This is my first issue. I am not sure what's the next step once PR is merged and pushed to prod https://github.com/Expensify/App/issues/19646#issuecomment-1577270722.


Since [the last "deployed to production" message](https://github.com/Expensify/App/issues/19646#issuecomment-1591975018) states payment should be due today I think we just need to wait on @michaelhaxhiu for compensation

**Comment by michaelhaxhiu** [comment]
Ok so just to be clear... there was 1 regression is that right? I am asking so that I can sort out the correct payment amounts.

**Comment by PauloGasparSv** [comment]
> Ok so just to be clear... there was 1 regression is that right? I am asking so that I can sort out the correct payment amounts.

Yes @michaelhaxhiu, we had 1 regression mentioned [here](https://github.com/Expensify/App/issues/19646#issuecomment-1581349596) and fixed by https://github.com/Expensify/App/pull/20462!

**Comment by michaelhaxhiu** [comment]
Ok so the payout would be:

- contributor = $1000
- C+ = $500 (50% dock for regression)

**Comment by michaelhaxhiu** [comment]
paid.
